corehq/ex-submodules/pillowtop/pillow/interface.py

```diff
@@ -371,7 +371,7 @@ def __record_change_metric_in_datadog(self, metric, change, processor=None, proc
             change_lag = (datetime.utcnow() - change.metadata.publish_timestamp).total_seconds()
             datadog_gauge('commcare.change_feed.change_lag', change_lag, tags=[
                 'pillow_name:{}'.format(self.get_name()),
-                _topic_for_ddog(change.topic),
+                _topic_for_ddog(change.partition if change.partition else change.topic),
```

Comment:
Will there no longer be a way to track by topic at all, or it is still there, just no longer the default? It might be nicer to add a new tag, so that we have name, topic, and partition.

Either way, I do not think this is the right function to use to format partition info for datadog. `_topic_for_ddog` expects a topic to be passed in and formats a tag for datadog called `topic`. If we want to send partition info, it would be good to have the tag called partition.
Something like 
`'partition:{}'.format(change.partition)`
 should be fine. However reading more of this, I think you actually may want to pass a `TopicPartition` which would encode both info. `_topic_for_ddog` is set up to handle that so something like
`_topic_for_ddog(TopicPartition(change.topic, change.partition))` may be ideal

Reply:
Right, so if we want to be able to group by topic in one graph and by topic-partition in another, then we'll want to have a separate `partition` tag like you suggested.

Like you said `_topic_for_ddog` does accept a number of different datatypes, one being the `TopicPartition`. My understanding was that `message.partition` in the function where `change.partition` is set is itself a `TopicPartition` (and not, say, an `int` or a `str`). BUT your comment made me double-check my logic and whadda ya know I was wrong. I'll update it to use `_topic_for_ddog(TopicPartition(change.topic, change.partition))` like you suggested.

Thanks for checking me! I have to say it can be pretty tough to trace back the type of things as they get passed around through this code, so in the end to test I had to just pull up a shell and see.

Reply:
Offlined with Cal, and in the end we decided to stick with just one `topic:` tag that's the topic-partition if there's a partition or else just the topic. It's in line with what some other metrics do, and having topic by itself doesn't mean much for this metric.